### DISCUSSION TITLE
How to consolidate Prisma migrations?

### DISCUSSION THREAD

**Discussion by mkastner** [keystonejs/keystone#8263] [Questions]
Either I am super sloppy or I am doing something wrong.

I've started a keystone project two weeks ago. While working on the project I've gathered more than fifty migrations so far. Restructuring, several typos, improved naming new feature requests led to this truckload of migrations and I guess a lot more are waiting around the corner.

Once in a while I'd like to check in a consolidated version of all migrations as single initial migration.

Is there a way to do this?

**Comment by dcousens** [comment]
@mkastner you have a few paths here, but maybe the better question to ask is, why?

Please ensure you read ["About migration histories"](https://www.prisma.io/docs/concepts/components/prisma-migrate/migration-histories) from Prisma first, but ignoring that, you could checkout your project locally at commit `xxx123`, setup your database, then checkout the newest commit `yyy345`, and write a migration for that jump.

Any attempt to change your migration history will probably result in Prisma errors, and you should only do this if you are willing to resolve the problems that come up.  If the contents of your database aren't important, you could delete and start fresh from your current schema.
